source/common/network/win32_socket_handle_impl.cc

```diff
@@ -71,10 +79,39 @@ Api::IoCallUint64Result Win32SocketHandleImpl::recvmmsg(RawSliceArrays& slices,
 }
 
 Api::IoCallUint64Result Win32SocketHandleImpl::recv(void* buffer, size_t length, int flags) {
-
-  Api::IoCallUint64Result result = IoSocketHandleImpl::recv(buffer, length, flags);
-  reEnableEventBasedOnIOResult(result, Event::FileReadyType::Read);
-  return result;
+  if (flags & MSG_PEEK) {
+    // can a remote OOM us now that we are not protected by readDisable?
+    Api::IoCallUint64Result peek_result = drainToPeekBuffer();
```

Comment:
I thought that the expectation was the reader should drain all the bytes from the wire in a while loop when reading similar to what `IoResult RawBufferSocket::doRead` does. I thought that if we don't call it until it blocks then we won't have the read events activated again.

Now that I think about it again it seems alright because the `connection_impl` will re-activate the read and the inspectors would be responsible for that.